Swift/Enum.swift.twig

```diff
@@ -8,8 +8,8 @@ import Foundation
 {% for value in values -%}
 /// - {{ utils.decapitalize(value.name) }}: {{ value.description }}
 {% endfor -%}
-enum {{ name }}: {{ enumutils.enumType(valuesTypes) }}, RawRepresentable {
+enum {{ name }}: {{ enumutils.enumType(valuesTypes) }}, Codable, RawRepresentable {
```

Comment:
please add `EnumeratedType` to LeadKit
```swift
protocol EnumeratedType {

    /// All possible options of type.
    static var allOptions: [Self] { get }

}

```

and implement it for all enums:

```swift
extension DateRange: EnumeratedType {

    static let allOptions: [DateRange] = [.today, .yesterday, .month, .threeMonths]

}
```

ref https://github.com/TouchInstinct/api-generator/issues/93

Reply:
and then include this in templates?